### PR TITLE
Django pattern library

### PR DESCRIPTION
**Pull request by William-Blackie**
- Adds Django pattern library as well as some basic button styling for testing purposes. 
- Refactors base.html and adds base_page.html to allow for non-page templates to have a wrapper for pattern-lab.
- Adds some basic styling for the new design, not yet complete but is to be built from.
- Adds basic styling for rich text.

### CODE REVIEW

File: wagtailio/settings/dev.py [hunk 1]
```diff
@@ -1,16 +1,24 @@
...
+# Display sent emails in the console while developing locally.
 EMAIL_BACKEND = "django.core.mail.backends.console.EmailBackend"
 
+# Use dummy app ID for development
 FB_APP_ID = 0
 
+# Do not force HTTP->HTTPS redirect when running the production setup on localhost
 SECURE_SSL_REDIRECT = False
 
+# Enable FE component library
```
Comment (William-Blackie) [inline wagtailio/settings/dev.py:19]: Note: shout if you want me to remove this, It's just a pet peeve when settings aren't fully documented.

File: wagtailio/project_styleguide/templates/patterns/atoms/svgs/arrow.html [hunk 2]
```diff
@@ -0,0 +1,3 @@
+<svg class="{% if classes %}{{ classes }}{% endif %}" viewBox="0 0 16 11" xmlns="http://www.w3.org/2000/svg" aria-hidden="true">
```
Comment (zerolab) [inline wagtailio/project_styleguide/templates/patterns/atoms/svgs/arrow.html:1]: nitpick: `patterns/atoms/svgs/` -> `patterns/atoms/icons/`
especially when you've got `static/sass/atoms/icons.scss` ;)

File: wagtailio/project_styleguide/templates/patterns/base.html [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+                {% with self.get_site.site_name as site_name %}
+                    {% if site_name %}- {{ site_name }}{% endif %}
+                {% endwith %}
+            {% endblock %}
+        </title>
+        <meta name="description" content="" />
+        <meta name="viewport" content="width=device-width, initial-scale=1" />
+
+        {# Global stylesheets #}
+        <link rel="stylesheet" type="text/css" href="{% static 'css/main.css' %}">
```
Comment (zerolab) [inline wagtailio/project_styleguide/templates/patterns/base.html:21]: nitpick: there is no more need for the default types for stylesheet and script tags

File: wagtailio/project_styleguide/templates/patterns/components/buttons/button.html [hunk 4]
```diff
@@ -0,0 +1,6 @@
+<a href="{{ url }}" class="button {% if classes %}{{ classes }}{% endif %}">
+    <p class="button__text">{{ title }}</p>
+    {% if arrow %}
+        {% include "patterns/atoms/svgs/arrow.html" with classes="arrow" %}
```
Comment (zerolab) [inline wagtailio/project_styleguide/templates/patterns/components/buttons/button.html:4]: wasn't the idea of `{% include "includes/svg_sprites.html" %}` to include all svg icons, so you can just reference them here?

File: wagtailio/project_styleguide/templates/patterns/atoms/svgs/arrow.html (hunk 2 above)
Comment (William-Blackie) [reply wagtailio/project_styleguide/templates/patterns/atoms/svgs/arrow.html:1]: Good point, I've pulled this out from your further point anyway.

File: wagtailio/project_styleguide/templates/patterns/components/buttons/button.html (hunk 4 above)
Comment (William-Blackie) [reply wagtailio/project_styleguide/templates/patterns/components/buttons/button.html:4]: It was, I was thinking of taking a mixed approach but realised realistically we're doubling effort. I've added the SVG to the sprites file instead.

File: wagtailio/project_styleguide/templates/patterns/base.html [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+            {% block title %}
+                {% if self.seo_title %}{{ self.seo_title }}{% else %}{{ self.title }}{% endif %}
+            {% endblock %}
+            {% block title_suffix %}
+                {% with self.get_site.site_name as site_name %}
+                    {% if site_name %}- {{ site_name }}{% endif %}
+                {% endwith %}
+            {% endblock %}
+        </title>
+        <meta name="description" content="" />
```
Comment (chris-lawton) [inline wagtailio/project_styleguide/templates/patterns/base.html:17]: Question: where will the content for this be pulled from? It looks like the existing site uses: ```{% if page.social_text %}{{ page.social_text }}{% elif page.listing_intro %}{{ page.listing_intro }}{% endif %}```

File: wagtailio/project_styleguide/templates/patterns/base.html [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+        <meta name="viewport" content="width=device-width, initial-scale=1" />
+
+        {# Global stylesheets #}
+        <link rel="stylesheet" href="{% static 'css/main.css' %}">
```
Comment (chris-lawton) [inline wagtailio/project_styleguide/templates/patterns/base.html:21]: we switched this to use manifest loader https://github.com/wagtail/wagtail.org/pull/175/commits/277de0fa91e559bbf3d693e3a7fac91b6e599000

`<link rel="stylesheet" href="{% manifest 'css/main.css' %}">`

You'll need to add the `manifest` tag as well

File: wagtailio/project_styleguide/templates/patterns/base.html [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+
+    <body class="{% block body_class %}{% endblock %}">
+        {% wagtailuserbar %}
+
+        {% block content %}
+            {{ pattern_library_rendered_pattern }}
+        {% endblock %}
+
+        {# Global javascript #}
+        <script src="{% static 'js/main.js' %}"></script>
```
Comment (chris-lawton) [inline wagtailio/project_styleguide/templates/patterns/base.html:39]:
```suggestion
        <script src="{% manifest 'js/main.js' %}"></script>
```

File: wagtailio/project_styleguide/templates/patterns/base_page.html [hunk 8]
```diff
@@ -1,4 +1,6 @@
-{% load static wagtailuserbar wagtailcore_tags wagtailimages_tags wagtail_ab_testing_tags wagtailio_utils manifest %}
+{% extends "patterns/base.html" %}
+{% load static wagtailcore_tags wagtailimages_tags wagtail_ab_testing_tags wagtailio_utils manifest %}
```
Comment (chris-lawton) [inline wagtailio/project_styleguide/templates/patterns/base_page.html:2]: `manifest` can be removed now the css and js files have been moved

File: wagtailio/project_styleguide/templates/patterns/components/buttons/button.html [hunk 9]
```diff
@@ -0,0 +1,6 @@
+<a href="{{ url }}" class="button {% if classes %}{{ classes }}{% endif %}">
+    <p class="button__text">{{ title }}</p>
+    {% if arrow %}
+        <svg class="arrow" ><use xlink:href="#arrow"></use></svg>
```
Comment (chris-lawton) [inline wagtailio/project_styleguide/templates/patterns/components/buttons/button.html:4]: nitpick:
```suggestion
        <svg class="arrow"><use xlink:href="#arrow"></use></svg>
```

File: wagtailio/settings/base.py [hunk 10]
```diff
@@ -218,8 +219,9 @@
 TEMPLATES = [
     {
         "BACKEND": "django.template.backends.django.DjangoTemplates",
-        "DIRS": [join(PROJECT_ROOT, "templates")],
-        "APP_DIRS": True,
+        "DIRS": [
+            os.path.join(PROJECT_ROOT, "project_styleguide/templates"),
+        ],        "APP_DIRS": True,
```
Comment (chris-lawton) [inline wagtailio/settings/base.py:224]: question: should `APP_DIRS` be on a new line?

File: wagtailio/project_styleguide/templates/patterns/base.html (hunk 5 above)
Comment (William-Blackie) [reply wagtailio/project_styleguide/templates/patterns/base.html:17]: so, in this case, `self` is whatever object (an instance page model in this case) calls it. So `self == page`. Just as an aside, this is set within the page's `get_context()` method, you can see this [here](https://github.com/wagtail/wagtail/blob/dc610366146b2dab04895d7619318d391ba435f1/wagtail/models/__init__.py#L1658), this is called when the page is rendered.

File: wagtailio/project_styleguide/templates/patterns/base.html (hunk 6 above)
Comment (William-Blackie) [reply wagtailio/project_styleguide/templates/patterns/base.html:21]: Rats, you are right! good catch.

File: wagtailio/project_styleguide/templates/patterns/base_page.html (hunk 8 above)
Comment (William-Blackie) [reply wagtailio/project_styleguide/templates/patterns/base_page.html:2]: :+1:

File: wagtailio/project_styleguide/templates/patterns/base.html (hunk 5 above)
Comment (chris-lawton) [reply wagtailio/project_styleguide/templates/patterns/base.html:17]: Should we add tags within `content=""` then? E.g.

```
<meta name="description" content="{% if page.social_text %}{{ page.social_text }}{% elif page.listing_intro %}{{ page.listing_intro }}{% endif %}" />
```

**Pull request merged**